src/framework/iron/api/kernel/iron_repository_factory.e

```diff
@@ -22,7 +23,10 @@ feature -- Factory
 					create iri.make_from_string (a_location)
 					create {IRON_WORKING_COPY_REPOSITORY} Result.make_from_uri (iri.to_uri)
 				else
-					create {IRON_WORKING_COPY_REPOSITORY} Result.make (a_location)
+					create l_wc_repo.make (a_location)
```

Comment:
Probably a good idea, didn't know it didn't do that. But for now something is better than nothing. Just no output when it doesn't do anything or will cause an error later is not good.

Reply:
Indeed, now (in my last commits) this verification is done a bit later. For instance, when we try to do `iron update`.

Also, registering a non-existing local working-copy repository may sound silly, but if you register a valid folder ... then for any reason it is deleted. Then it is better to do the check when we try to use it.

I also added `iron repository --clean` that will unregister invalid repository.
I was tempted to do also the check for remote server, but it can be only a temporary failure for remote server. So for now, the "cleaning" is done mostly for local repository.

Reply:
I think I still also prefer the check when registering, i.e. when you're playing around with the tool for the 1st time, these are all issues you bump into. It's all nice when everything is working and you understand it, but if you don't, better to be clear at the point you're doing something that doesn't make sense. I.e. fail early, not later.

Reply:
I agree, I've just committed and pushed a change to check , at registration time, if the location is valid.
For now, there is only an implementation for the working-copy repositories.